### PR TITLE
open option for lock

### PR DESCRIPTION
**Pull request by bkbilly**
## Proposed change

Add an extra button on more-info dialog to open the door.

image

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/frontend/discussions/14821
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by karwosts** [comment]
Is there any way to find out if a lock supports this before displaying the button?

I would assume most locks don't support this, and I would find it kind of confusing in that case.

**Comment by bkbilly** [comment]
I did a bit of research and I found out that there are supported_features which include if the door supports open.
I am going to try to implement this week and I'll upload it.

**Comment by koulis-probotek** [comment]
image

Show open button only if it supported by the lock.
Also fixed a bug with the hidden attribute which helps hide the lock or unlock if it already on that state.

**Comment by bkbilly** [comment]
image

Show open button only if it supported by the lock.
Also fixed a bug with the hidden attribute which helps hide the lock or unlock if it already on that state.

**Comment by bkbilly** [comment]
Could someone review this?
I would love to have the open option on the frontend.

**Comment by steffenrapp** [comment]
PR #15995 also seems to include the open option.

File: src/state-summary/state-card-lock.js [hunk 1]
```diff
@@ -61,12 +71,14 @@ class StateCardLock extends LocalizeMixin(PolymerElement) {
...
       },
       isLocked: Boolean,
+      supportsOpen: Boolean,
     };
   }
 
   _stateObjChanged(newVal) {
     if (newVal) {
       this.isLocked = newVal.state === "locked";
+      this.supportsOpen = supportsFeature(newVal, 1);
```
Comment (bramkragten) [inline src/state-summary/state-card-lock.js:81]: Please store the supported features in a `data/lock.ts` file like this:
https://github.com/home-assistant/frontend/blob/dev/src/data/fan.ts#L17

Comment (bkbilly) [reply src/state-summary/state-card-lock.js]: Thanks for the help, I've pushed an update with this feature.

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**